FormalSystems/Preliminaries/Language.lean

```diff
@@ -227,7 +234,13 @@ theorem Sigma.kleene_contains_all : ∀(w : Word α), w ∈ (Σ)∗
   | [] => by exists 0
   | x::xs => by
     have ⟨n, hn⟩ := Sigma.kleene_contains_all xs
-    exists n + 1; exists [x]; exists xs
+    exists (n + 1)
+    rw [pow_succ] 
+    conv => right; right; rw [← pow_one (Σ)]
+    rw [npow_mul_comm]
+    simp
+    exists [x]
+    exists xs
```

Comment:
This part took me some time. There might be an easier way to do this (e.g. by using a theorem from mathlib that I'm not aware of).

Reply:
Here is a small simplification. I guess the problem here is, that the simplification performed by `exists` doesn't work anymore (which is likely caused by the changes to the simp tactic).

```suggestion
    exists (1 + n)
    simp [pow_succ]
    exists [x]
    exists xs
```

Reply:
Ah, I tried to use `(1+n)` but gave up since the rest did not work immediately. Thanks!